components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementAdminService.java

```diff
@@ -49,14 +51,16 @@
      * user will assigned to the created role.Internal roles used.
      *
      * @param serviceProvider Service provider
+     * @param spTemplateContent SP template XML content
      * @return application id
      * @throws org.wso2.carbon.identity.application.common.IdentityApplicationManagementException
      */
-    public void createApplication(ServiceProvider serviceProvider) throws IdentityApplicationManagementException {
-
+    public void createApplication(ServiceProvider serviceProvider, String spTemplateContent)
+            throws IdentityApplicationManagementException {
```

Comment:
Are we removing the API createApplication(ServiceProvider serviceProvider) ? Is it correct?

Reply:
createApplication won't be removed, considered as deprecated and will be calling addApplication from that.